core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java

```diff
@@ -846,17 +848,13 @@ public void execute(final BuildContext bc) {
                                 }
                                 Object result;
                                 try {
-                                    result = method.invoke(instance, methodArgs);
+                                    result = lookup.unreflect(method).bindTo(instance).invokeWithArguments(methodArgs);
                                 } catch (IllegalAccessException e) {
                                     throw ReflectUtil.toError(e);
-                                } catch (InvocationTargetException e) {
-                                    try {
-                                        throw e.getCause();
-                                    } catch (RuntimeException | Error e2) {
-                                        throw e2;
-                                    } catch (Throwable t) {
-                                        throw new IllegalStateException(t);
-                                    }
+                                } catch (RuntimeException | Error e2) {
+                                    throw e2;
+                                } catch (Throwable t) {
+                                    throw new IllegalStateException(t);
```

Comment:
I generally use `UndeclaredThrowableException` for these.

Reply:
Done